### PR TITLE
Katex

### PR DESCRIPTION
**Pull request by Hoverbear**
As requested in #5, this adds KaTeX support to our mdbook via a theme override.

Follows the path of https://github.com/rust-lang-nursery/mdBook/issues/49#issuecomment-167002731.

Add latex like so:

## Blocks

```
\\[ Your math here... \\]
```

## Inline

```
\\( Your math here... \\)
```

### CODE REVIEW

**Comment by siddontang** [comment]
@Hoverbear 

how about Gitbook, maybe we will generate the book for Gitbook too.

**Comment by Hoverbear** [comment]
@siddontang We are currently not using Gitbook, and as far as I am aware have no intention to. I think introducing Gitbook is a different issue entirely and I think it is a waste of time, `mdbook` accomplishes the same task.

File: book/index.html [hunk 1]
```diff
@@ -78,7 +78,15 @@
         <div id="page-wrapper" class="page-wrapper">
 
             <div class="page">
-                
+                <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.css" integrity="sha384-D+9gmBxUQogRLqvARvNLmA9hS2x//eK1FhVb9PiU86gmcrBrJAQT8okdJ4LMp2uv" crossorigin="anonymous">
```
Comment (siddontang) [inline book/index.html:81]: is the link generated automatically by mdbook?

Comment (Hoverbear) [reply book/index.html:81]: Everything in the `book/` folder is an automated build output, please don't review it.

The only actual change is introduce this file: `theme/header.hbs`

Comment (Hoverbear) [reply book/index.html:81]: The content of `theme/header.hbs` is the reccomended way to deploy KaTeX. The LaTeX/math renderer made by Khan Academy which is very very good.

**Review by siddontang — APPROVED** [review]
LGTM

**Comment by siddontang** [comment]
PTAL @DorianZheng

**Comment by DorianZheng** [comment]
@Hoverbear I am confused. The issue you mentioned in description looks like talking about adding `MathJax` to support formula, while this PR introduce `KaTeX`. Is it the same thing?

**Comment by Hoverbear** [comment]
@DorianZheng It is a different tool to accomplish the same task. KaTeX is faster and better supported.

**Comment by siddontang** [comment]
PTAL @DorianZheng

**Comment by overvenus** [comment]
```diff
diff --git a/book.toml b/book.toml
index ca731db..634be1c 100644
--- a/book.toml
+++ b/book.toml
@@ -4,5 +4,5 @@ multilingual = false
 src = "src"
 title = "Head First TiKV"

-[output.html]
-theme = "theme"
+# [output.html]
+# theme = "theme"
diff --git a/src/key-value-engine/B-Tree_vs_The_Log-Structured_Merge-Tree.md b/src/key-value-engine/B-Tree_vs_The_Log-Structured_Merge-Tree.md
index b10d41d..988ae67 100644
--- a/src/key-value-engine/B-Tree_vs_The_Log-Structured_Merge-Tree.md
+++ b/src/key-value-engine/B-Tree_vs_The_Log-Structured_Merge-Tree.md
@@ -42,6 +42,8 @@ In the B+ tree, copies of the keys are stored in the internal nodes; the keys an

 To simplify the analysis, assume that the block size of the tree is `B`, and keys, pointers, and records are constant size, so that each internal node contains `O(B)` children and each leaf contains `O(B)` data records. (The root node is a special case, and can be nearly empty in some situations.) Under all these assumptions, the depth of a B tree is

+\\[ O(log_{B}\frac{N}{B}) \\]
+
 <center><code>O(log<sub>B</sub>N/B)</code></center>

 where `N` is the size of the database.
```

~~After commenting out `theme`, katex still works.~~ (forgot to remove `theme` dir.)

screenshot_20181011_134934

```
$ mdbook --version
mdBook v0.2.1
```

**Comment by siddontang** [comment]
cool

**Comment by Hoverbear** [comment]
Oh, wow, cool. Let me clean that then. :)

**Comment by siddontang** [comment]
PTAL @overvenus

**Review by overvenus — APPROVED** [review]
:+1: LGTM

**Pull request merged**